core/src/main/scala/za/co/absa/spline/harvester/builder/UnionNodeBuilder.scala

```diff
@@ -35,34 +36,49 @@ class UnionNodeBuilder
   override lazy val functionalExpressions: Seq[FunctionalExpression] =
     unionInputs
       .zip(operation.output)
-      .map{ case (input, output) => constructUnionFunction(input, output)}
+      .map { case (input, output) => constructUnionFunction(input, output) }
 
   override lazy val outputAttributes: Seq[Attribute] =
     unionInputs
       .zip(functionalExpressions)
-      .map{ case (input, function) => constructUnionAttribute(input, function)}
+      .map { case (input, function) => constructUnionAttribute(input, function) }
 
   private def constructUnionFunction(
     inputSplineAttributes: Seq[Attribute],
     outputSparkAttribute: sparkExprssions.Attribute
   ) =
     FunctionalExpression(
-      id =  UUID.randomUUID().toString,
+      id = UUID.randomUUID().toString,
       dataType = componentCreatorFactory.dataTypeConverter
         .convert(outputSparkAttribute.dataType, outputSparkAttribute.nullable).id.asOption,
       childRefs = inputSplineAttributes.map(att => AttrOrExprRef(Some(att.id), None)).asOption,
-      extra = Map("synthetic" -> "true").asOption,
-      name = "union",
-      params = Map("name" -> "union").asOption
+      extra = Map(ExtraFields.Synthetic -> true).asOption,
+      name = Names.Union,
+      params = None
     )
 
-  private def constructUnionAttribute(inputSplineAttributes: Seq[Attribute], function: FunctionalExpression) =
+  private def constructUnionAttribute(attributes: Seq[Attribute], function: FunctionalExpression) = {
+    val attr1 +: attrs = attributes
+    assume(attrs.forall(_.name == attr1.name))
```

Comment:
This assumption is wrong. Spark doesn't require the name to be the same. 

Spark docs for UNION:
> Note that input relations must have the same number of columns and compatible data types for the respective columns.